cpp_code/tests/test_random_number_generator.cpp

```diff
@@ -249,43 +279,66 @@ static int random_seed(void) {
 }
 
 static void test_uniform_integer(RandomNumberGenerator &rng) {
-    vector<size_t> counts(PSI_DF);
     vector<double> probabilities(PSI_DF);
     size_t i;
+    unsigned passes, trials;
 
     for (i = 0; i < probabilities.size(); i++)
         probabilities[i] = 1/static_cast<double>(PSI_DF);
-    for (i = 0; i < NSAMPLES; i++)
-        counts[rng.nexti(PSI_DF)]++;
-    assert(psi_test(counts, probabilities, NSAMPLES));
+
+    for (passes = 0, trials = NTRIALS; trials --> 0;) {
```

Comment:
If you ever use `trials` in the body of the loop as an index into an array of length `NTRIALS`, then `trials > 0; --trials` is wrong.  I thought the arrow was ridiculous when I first saw it, but I've come to the conclusion that it is the best way to write reverse iteration in C: it is memorable and practically guaranteed to be the right thing without thinking.  (I used to write `for (x = N; 0 < x--;)`, since I like writing things to be ascending left to right, but this arrow trick is clearer.)